src/storage/memory_store.ts

```diff
@@ -300,6 +361,15 @@ export class MemoryStore implements IStore {
    * @returns {Promise<void>}
    */
   async saveTx(tx: IHistoryTx): Promise<void> {
+    // Protect ordering list from updates on the same transaction
+    // We can check the historyTs but it's O(n) and this check is O(1).
+    if (!this.history.has(tx.tx_id)) {
+      // Add transaction to the ordering list
+      // and sort it so we ensure the order
+      this.historyTs.push(getOrderingKey(tx));
+      this.historyTs.sort();
```

Comment:
I would only add a comment here explaining why we need this list sorted by timestamp and that, if this sort becomes a performance problem in the future, that we can create a pre processed data for each token and remove this sort.